src/option.c

```diff
@@ -537,3 +539,83 @@ coap_option_filter_get(const coap_opt_filter_t filter, uint16_t type) {
    * but as _set and _unset do, the function does not take a const). */
   return coap_option_filter_op((uint16_t *)filter, type, FILTER_GET);
 }
+
+coap_optlist_t *coap_new_optlist(uint16_t number,
+                                 size_t length,
+                                 const uint8_t *data) {
+  coap_optlist_t *node;          
+     
+  node = coap_malloc_type(COAP_OPTLIST, sizeof(coap_optlist_t) + length);
+     
+  if (node) {
+    node->number = number;
+    node->length = length;
+    memcpy(node->data, data, length);
+  } else {
+    coap_log(LOG_DEBUG, "coap_new_optlist: malloc failure\n");
```

Comment:
Is `LOG_DEBUG` a good log level here? Maybe something like `LOG__WARNING` could be more appropriate?

Reply:
Fixed in Update to be LOG_WARNING